common/flagdproxy/flagdproxy.go

```diff
@@ -62,32 +64,44 @@ func (f *FlagdProxyHandler) Config() *FlagdProxyConfiguration {
 	return f.config
 }
 
+func (f *FlagdProxyHandler) createObject(ctx context.Context, obj client.Object) error {
+	return f.Client.Create(ctx, obj)
+}
+
+func (f *FlagdProxyHandler) updateObject(ctx context.Context, obj client.Object) error {
+	return f.Client.Update(ctx, obj)
+}
+
 func (f *FlagdProxyHandler) HandleFlagdProxy(ctx context.Context) error {
```

Comment:
what should we do if a user deploys manually a `flagd-proxy` instance using the same name and in the same namespace but the `.spec` differs, i.e., the version is different? Currently, we just update it but IMHO we should look at the `managed-by` label. I am thinking about the case where there is a GitOps system in place (eg Argo) that deploys the proxy to a different version comparing to what OFO expects. This would trigger a continuous loop between OFO changing the deployment and Argo changing it back 🤔

Reply:
yes that's true; not sure if `managed-by` label is set when installing from manifests; but we can definitely look at the `ownerReferences`. This should be set all the time -> Here we need to adapt the behavior a little bit (if we cannot fetch OFO Deployment -> do not create/update `flagd-proxy`)

WDYT?

Reply:
`ownerReferences` is meant for GC, I don't know if it would fit this use case. Furthermore, `ownerReferences` would differ between two OFO versions, name and uid fields are not preserved between updates.

Reply:
adapted according to description

Reply:
I think this is an interesting point... either a label, or maybe a special annotation?

Reply:
adapted it to use a label to check if the flagd-proxy deployment was deployed by OFO, can we maybe align here on the name of the label? @toddbaert @thisthat @Kavindu-Dodan ?

I would suggest 

`openfeature.dev/deployed: "true"`

WDYT?

Reply:
Agreed with @thisthat to used `managed-by` k8s annotation, adapted according to this

Reply:
I think the new update to use `managed-by` is a great improvement 👍

Reply:
@toddbaert if you agree with the solution, we should be ready to merge